app/controllers/user_targets_controller.rb

```diff
@@ -38,4 +39,31 @@ def load_target
   def update_first_target
     current_user.update(first_target: true) unless current_user.first_target
   end
+
+  def check_new_matches(new_target)
+    new_target_coord = Geokit::LatLng.new(new_target.latitude,new_target.longitude)
+
+    new_matches = UserTarget.within(new_target.area, origin: new_target_coord).not_from_user(new_target.user_id).with_topic(new_target.topic_id)
+
+    new_matches.each do |target|
+      second_user = target.user
+      match = Match.create(first_target: new_target, second_target: target)
+      UserMatch.create(user: current_user, match: match)
+      UserMatch.create(user: second_user, match: match)
+      send_new_match_notification(second_user, new_target.user.name, '1')
+    end
+
+    new_matches_count = new_matches.size
+    if(new_matches_count >= 1)
```

Comment:
Follow the ruby guide 
https://github.com/rootstrap/guides/tree/master/ruby#if-as-a-modifier
https://github.com/rootstrap/guides/tree/master/ruby#no-parens-around-condition